assets/Product.wxs

```diff
@@ -130,4 +179,105 @@
+      <Control Id="BannerBitmap" Type="Bitmap" X="0" Y="0" Width="370" Height="44" TabSkip="no" Text="!(loc.InstallDirDlgBannerBitmap)"/>
+      <Control Id="BannerLine" Type="Line" X="0" Y="44" Width="370" Height="0"/>
+      <Control Id="BottomLine" Type="Line" X="0" Y="234" Width="370" Height="0"/>        
+    </Dialog>
+  </UI>
+</Fragment>
+
+<!-- Customized version of WixUI_InstallDir, which is necessary to add custom dialogs. https://github.com/wixtoolset/wix3/blob/master/src/ext/UIExtension/wixlib/WixUI_InstallDir.wxs -->
+<Fragment>
+  <UI Id="CustomWixUI_InstallDir">        
+      <!--
+      First-time install dialog sequence:
+      - WixUI_WelcomeDlg
+      - WixUI_LicenseAgreementDlg
+      - WixUI_InstallDirDlg
+      - ExplorerContextMenuDialog
+      - WixUI_VerifyReadyDlg
+      - WixUI_DiskCostDlg
+      Maintenance dialog sequence:
+      - WixUI_MaintenanceWelcomeDlg
+      - WixUI_MaintenanceTypeDlg
+      - WixUI_InstallDirDlg
+      - WixUI_VerifyReadyDlg
+      Patch dialog sequence:
+      - WixUI_WelcomeDlg
+      - WixUI_VerifyReadyDlg
+      -->
+
+      <TextStyle Id="WixUI_Font_Normal" FaceName="Tahoma" Size="8" />
+      <TextStyle Id="WixUI_Font_Bigger" FaceName="Tahoma" Size="12" />
+      <TextStyle Id="WixUI_Font_Title" FaceName="Tahoma" Size="9" Bold="yes" />
+
+      <Property Id="DefaultUIFont" Value="WixUI_Font_Normal" />
+      <Property Id="WixUI_Mode" Value="InstallDir" />
+
+      <DialogRef Id="BrowseDlg" />
+      <DialogRef Id="DiskCostDlg" />
+      <DialogRef Id="ErrorDlg" />
+      <DialogRef Id="FatalError" />
+      <DialogRef Id="FilesInUse" />
+      <DialogRef Id="MsiRMFilesInUse" />
+      <DialogRef Id="PrepareDlg" />
+      <DialogRef Id="ProgressDlg" />
+      <DialogRef Id="ResumeDlg" />
+      <DialogRef Id="UserExit" />
+      <DialogRef Id="ExplorerContextMenuDialog" />
+      
+      
+      <Publish Dialog="BrowseDlg" Control="OK" Event="DoAction" Value="WixUIValidatePath" Order="3">1</Publish>
+      <Publish Dialog="BrowseDlg" Control="OK" Event="SpawnDialog" Value="InvalidDirDlg" Order="4"><![CDATA[NOT WIXUI_DONTVALIDATEPATH AND WIXUI_INSTALLDIR_VALID<>"1"]]></Publish>
+
+      <Publish Dialog="ExitDialog" Control="Finish" Event="EndDialog" Value="Return" Order="999">1</Publish>
+
+      <Publish Dialog="WelcomeDlg" Control="Next" Event="NewDialog" Value="LicenseAgreementDlg">NOT Installed</Publish>
+      <Publish Dialog="WelcomeDlg" Control="Next" Event="NewDialog" Value="VerifyReadyDlg">Installed AND PATCH</Publish>
+
+      <Publish Dialog="LicenseAgreementDlg" Control="Back" Event="NewDialog" Value="WelcomeDlg">1</Publish>
+      <Publish Dialog="LicenseAgreementDlg" Control="Next" Event="NewDialog" Value="InstallDirDlg">LicenseAccepted = "1"</Publish>
+
+      <Publish Dialog="ExplorerContextMenuDialog" Control="Back" Event="NewDialog" Value="InstallDirDlg">1</Publish>
```

Comment:
This implementation of the Back button looks OK, does not have bug <https://github.com/PowerShell/PowerShell/issues/12055> yet.